lib/reporting/proofing_rate_report.rb

```diff
@@ -141,6 +130,43 @@ def cloudwatch_client
         wait_duration: wait_duration,
       )
     end
+
+    def trailing_days_ranges
+      [0, *DATE_INTERVALS].each_cons(2).map do |slice_end, slice_start|
+        time_range = if slice_end.zero?
+                       Range.new(
+                         (end_date - slice_start.days).beginning_of_day,
+                         (end_date - slice_end.days).end_of_day,
+                       )
+                     else
+                       Range.new(
+                         (end_date - slice_start.days).beginning_of_day,
+                         (end_date - slice_end.days).end_of_day - 1.day,
+                       )
+                     end
+        Reporting::IdentityVerificationReport.new(
+          issuers: nil, # all issuers
+          time_range: time_range,
+          cloudwatch_client: cloudwatch_client,
+        )
+      end
+    end
+
+    def by_month_ranges
+      ranges = [
+        end_date.last_month.last_month.all_month,
+        end_date.last_month.all_month,
+        end_date.all_month,
```

Comment:
I think we actually want these offset; e.g., the August 1 run of this should include May, June, and July. I should also almost certainly use date math here instead of chaining `.last_month` repeatedly. 😇